src/inmanta/deploy/state.py

```diff
@@ -204,6 +204,25 @@ def copy(self: Self) -> Self:
         """
         return dataclasses.replace(self)
 
+    def to_handler_state(self) -> const.ResourceState:
+        match self:
+            case ResourceState(compliance=Compliance.UNDEFINED):
+                return const.ResourceState.undefined
+            case ResourceState(blocked=Blocked.BLOCKED):
+                return const.ResourceState.skipped_for_undefined
+            case ResourceState(compliance=Compliance.HAS_UPDATE):
+                return const.ResourceState.available
+            case ResourceState(last_deploy_result=DeployResult.SKIPPED):
+                return const.ResourceState.skipped
+            case ResourceState(last_deploy_result=DeployResult.FAILED):
+                return const.ResourceState.failed
+            case ResourceState(compliance=Compliance.NON_COMPLIANT):
+                return const.ResourceState.non_compliant
+            case ResourceState(last_deploy_result=DeployResult.DEPLOYED):
+                return const.ResourceState.deployed
+            case _:
```

Comment:
Have you tried `typing.assert_never` here? I don't recall off the top of my head if mypy can handle it, but I think it might now that we extended it to cover all `Compliance` scenarios. If so, we don't need the catch-reraise in the caller.